src/Symfony/Component/Messenger/Tests/Transport/RedisExt/RedisExtIntegrationTest.php

```diff
@@ -55,11 +55,48 @@ public function testGetTheFirstAvailableMessage()
         $this->assertEquals(['type' => DummyMessage::class], $encoded['headers']);
     }
 
-    private function clearRedis()
+    public function testConnectionSendWithSameContent()
     {
-        $parsedUrl = parse_url(getenv('MESSENGER_REDIS_DSN'));
-        $pathParts = explode('/', $parsedUrl['path'] ?? '');
-        $stream = $pathParts[1] ?? 'symfony';
-        $this->redis->del($stream);
+        $body = '{"message": "Hi"}';
+        $headers = ['type' => DummyMessage::class];
+
+        $this->connection->add($body, $headers);
+        $this->connection->add($body, $headers);
+
+        $encoded = $this->connection->get();
+        $this->assertEquals($body, $encoded['body']);
+        $this->assertEquals($headers, $encoded['headers']);
+
+        $encoded = $this->connection->get();
+        $this->assertEquals($body, $encoded['body']);
+        $this->assertEquals($headers, $encoded['headers']);
+    }
+
+    public function testConnectionSendAndGetDelayed()
+    {
+        $this->connection->add('{"message": "Hi"}', ['type' => DummyMessage::class], 500);
+        $encoded = $this->connection->get();
+        $this->assertNull($encoded);
+        usleep(500000);
+        $encoded = $this->connection->get();
+        $this->assertEquals('{"message": "Hi"}', $encoded['body']);
+        $this->assertEquals(['type' => DummyMessage::class], $encoded['headers']);
+    }
+
+    public function testConnectionSendDelayedMessagesWithSameContent()
+    {
+        $body = '{"message": "Hi"}';
+        $headers = ['type' => DummyMessage::class];
+
+        $this->connection->add($body, $headers, 500);
+        $this->connection->add($body, $headers, 500);
+        usleep(500000);
```

Comment:
We should mark the test class as `@group time-sensitive`

Reply:
Added 👍